src/cljs/main/broadfcui/common/table/column_editor.cljs

```diff
@@ -0,0 +1,130 @@
+(ns broadfcui.common.table.column-editor
+  (:require
+    [dmohs.react :as react]
+    [broadfcui.common :as common]
+    [broadfcui.common.components :as comps]
+    [broadfcui.common.icons :as icons]
+    [broadfcui.common.overlay :as overlay]
+    [broadfcui.common.table.utils :as table-utils]
+    [broadfcui.common.style :as style]
+    [broadfcui.utils :as utils]
+    ))
+
+
+(defn- set-all [{:keys [column-display update-column-display fixed-column-count]} visible?]
+  (->> (drop fixed-column-count column-display)
+       (map #(assoc % :visible? visible?))
+       (concat (take fixed-column-count column-display))
+       vec
+       update-column-display))
+
+
+(react/defc ColumnEditor
+  (->>
+   {:render
+    (fn [{:keys [props state]}]
+      (let [{:keys [drag-index drop-index]} @state
+            {:keys [columns column-display update-column-display fixed-column-count dismiss]} props
+            reorderable-columns (vec (drop fixed-column-count column-display))]
+        [:div {:className (when drag-index "grabbing-icon")
+               :style {:border (str "2px solid " (:line-default style/colors))
+                       :padding "1em" :lineHeight "1.5em"}}
+         [:div {:style {:display "flex" :marginBottom "0.5em" :justifyContent "space-between"
+                        :padding "4px 8px" :borderRadius 5 :cursor (when-not drag-index "pointer")
+                        :border (str "1px solid " (:button-primary style/colors))
+                        :color (:button-primary style/colors)}
+                :onClick (fn [_]
+                           (update-column-display (table-utils/build-column-display columns))
+                           (dismiss))}
+          (icons/icon {:style {:fontSize 18 :lineHeight "inherit"}} :reset)
+          [:span {:style {:fontSize 14 :flexGrow 1 :textAlign "center"}}
+           "Reset Columns"]]
+         "Show:"
+         (let [style {:width "4rem" :padding "4px 8px" :marginRight "0.5rem" :borderRadius 5}]
+           [:div {:style {:padding "0.5em 0"}}
+            [comps/Button {:style style :onClick #(set-all props true) :text "All"}]
+            [comps/Button {:style style :onClick #(set-all props false) :text "None"}]])
+         (map-indexed
```

Comment:
I think fixed columns would be clearer to users if they showed up as immovable in this list.